src/metabase/api/card.clj

```diff
@@ -1150,6 +1150,7 @@ saved later when it is ready."
   (let [db-id             (public-settings/uploads-database-id)
         database          (or (t2/select-one Database :id db-id)
                               (throw (Exception. (tru "The uploads database does not exist."))))
+        _check_perms      (api/check-403 (mi/can-write? database))
```

Comment:
Based on the Slack thread I think this should be `mi/can-read?` instead. That will check that a user has data perms for the database. `can-write?` checks whether the user can update the database details in Metabase, which is a different type of permission.